nativelib/src/main/scala/scala/scalanative/runtime/ieee754tostring/ryu/RyuDouble.scala

```diff
@@ -699,21 +699,80 @@ object RyuDouble {
       roundingMode: RyuRoundingMode
   ): String = {
 
-    // Step 1: Decode the floating point number, and unify normalized and
-    // subnormal cases.
-    // First, handle all the trivial cases.
+    // Handle all the trivial cases.
```

Comment:
It is a matter of personal preference, style, and differing
estimates of how often NaN  and +/- Infinity are going
to be encountered.

I would tend to delete this trivial check and keep the wrapper as thin as feasible.  It is not wrong, just lengthens every `doubleToString` with a redundant check.

Reply:
Good point.

The motivation behind the duplication is to be sure that:
1. we don't do `new scala.Array[Char](_RyuDouble.RESULT_STRING_MAX_LENGTH)` for trivial cases
2. doubleToChars is doing these checks

What we could do is to have an intermediate `private def _doubleToCharsNoCheck()` that would be called by `doubleToString()` and `doubleToChars()`.
This way the checks would be performed only once in the public methods.

Reply:
> we don't do new scala.Array[Char](_RyuDouble.RESULT_STRING_MAX_LENGTH) for trivial cases.

As we so much of what we have been considering recently, so much depends upon workloads and we devos
have not real control over that. What is the expected ratio over all time & workloads of NaN, +/- Infinity versus
other IEEE7454 numbers, I hesitate to call them 'real' as in Pinocchio 'real boy' real for fear of confusion.
I wish I knew, or had the time to know, the compose sequence for the mathematicians symbol for the set of Real numbers.

Nan & friends need to be detected. I am willing to believe that the expected ration of Actual-to-trivial is almost
overwhelmingly in favor of the Actual numbers. To my thinking that path should be as fast/short as possible.

Nan and friends need to be detected and handled, but, to my thinking, they do not need to be any faster than any
Actual number if that slows down the "expected" path.  Yes, the doing allocations on the "trivial" path is not
strictly necessary. And Yes, one can easily design "pathological" workloads which will trigger these "unnecessary"
allocations.

Reply:
As always, said in a friendly manner, Your PR, your call. 

Getting the PR to "Done" is more important.